internal/configsource/includeconfigsource/README.md

```diff
@@ -1,19 +1,65 @@
 # Include Config Source (Alpha)
 
 Use the include config source to inject [golang templates](https://pkg.go.dev/text/template)
-into the configuration.
+or plain files into the configuration.
 
 ## Configuration
 
 Under the `config_sources:` use `include:` or `include/<name>:` to create a
 template config source.
+The following parameters are available to customize the include config sources:
 
 ```yaml
 config_sources:
   include:
+  include/my_name_00:
+    # delete_files can be used to make the "include" config source to delete the
+    # files referenced by it. This is typically used to remove secrets from the
+    # file system. The file is deleted as soon as its value is read by the config
+    # source. The default value is false. It is an invalid configuration to set it
+    # to true together with the watch_files parameter (see below).
+    delete_files: true
+  include/my_name_01:
+    # watch_files can be used to make the "include" config source monitor for
+    # updates on the used files. Setting it to true will trigger a configuration
+    # reload if any of the files used by the config source are updated.
+    # Configuration reload causes temporary interruption of the data flow during
+    # the time taken to shut down the current pipeline configuration and start the
+    # new one. The default value is false. It is an invalid configuration to set it
+    # to true together with the delete_files parameter (see above).
+    watch_files: true
 ```
 
-The parameters of a include config source are passed to template to be processed.
+Example on how to use the `delete_files` and `watch_files`:
+```yaml
+config_sources:
+  include/default:
+  include/secret:
+    delete_files: true
+  include/watch_for_updates:
+    watch_files: true
+
+components:
+  component_0:
+    # Default usage: configuration won't be reloaded if the file
+    # '/etc/configs/component_field' is changed.
+    component_field: ${include/default:/etc/configs/component_field} 
```

Comment:
If it's possible to use the contents of a yml file for the complete mapping, including the specified components, I think an example below would be helpful for users who want to do something like

```
receivers:
  ${include/default:/etc/receivers.yaml}

pipelines:
  ${include/default:/etc/pipelines.yaml}
```

Reply:
Added an example to cover this case.